pkg/loader/loader.go

```
@@ -88,6 +94,12 @@ func (l *loader) Parse(ctx context.Context, progReader io.ReaderAt) (*ParsedELF,
 		return nil, err
 	}
 
+	for _, prog := range spec.Programs {
+		if prog.Type == ebpf.UnspecifiedProgram {
```

just out of curiosity did you run into a situation where this check was needed?

---

I did! When running a BPF program that doesn't specify a `SEC` according to https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/tree/tools/lib/bpf/libbpf.c#n9002, it will silently fail to load the program. This line ensures there's at least a debug log for a user to figure out why their program isn't being included.